app/controllers/group/archive_controller.rb

```diff
@@ -10,13 +10,7 @@ class Group::ArchiveController < ApplicationController
   before_action :authorize_action
 
   def create
-    ActiveRecord::Base.transaction do
-      archival_timestamp = Time.zone.now
-
-      archive_roles(archival_timestamp) and
-        archive_group(archival_timestamp) and
-        entry.save!
-    end
+    entry.archive!
```

Comment:
Normalerweise wird in Rails glaube ich empfohlen, Business-Logik eher in Controllern als in Models zu machen. Hast du einen konkreten Grund warum du diese Logik vom Controller ins Model verschoben hast?

Reply:
Konkret habe ichs dann vorallem für Specs weiterverwendet und wollte es DRY halten.
Ich finde aber auch, dass `archive!` sich für mich sehr sinnvoll bei zB. `destroy!` einreiht. Es macht für mich Sinn auf dem Model eine Methode zum archivieren zu haben. 
Mit einer Domain Klasse `Group::Archiver` beispielsweise könnte mans aber auch DRY haben und hätte die Logik voll getrennt. Fänd ich jetzt hier nicht unbedingt nötig aber vielleicht siehst du das anders :thinking:

Reply:
Passt für mich so, den Vergleich mit `destroy!` finde ich sehr passend.